### DISCUSSION TITLE
How to get access to Roslyn Syntax Tree (or similar) in a MEF component?

### DISCUSSION THREAD

**Discussion by Hyperwavez** [VsixCommunity/Community.VisualStudio.Toolkit#348] [Q&A]
In short, I'm trying to get access to the code context of the caret position in a KeyProcessor component.

For example:
- Check if the caret position is in the scope of a namespace, class, method, or loop mechanic
- Check if the caret position is between the quotation marks of a string litteral

I have access to the IWpfTextView interface by default, but I'm not sure where to go from there.
I read somewhere that the IBufferGraph interface has similar functionality as the Roslyn Syntax Tree, but I don't understand how to use it for that purpose.

Any ideas?

**Comment by madskristensen** [comment]
Not sure how to do that. Perhaps @jaredpar knows where to get started?

**Comment by reduckted** [comment]
`VisualStudioWorkspace` is the class that you need to import to get access to the Roslyn workspace that Visual Studio is using. 

```cs
Import(typeof(VisualStudioWorkspace))]
public VisualStudioWorkspace Workspace { get; set; }
```

From there, you can use Roslyn methods to get the token at a position, etc, etc.

I think you will need to add the [`Microsoft.VisualStudio.LanguageServices`](https://www.nuget.org/packages/Microsoft.VisualStudio.LanguageServices/) NuGet package to access that type.

**Reply by Hyperwavez** [reply]
Thank you! This looks like it should do the trick. 
I haven't been able to test it yet so I'll leave this open for now, but I'll see if I get it working later today.
